src/Compilers/Core/CodeAnalysisTest/AnalyzerAssemblyLoaderTests.cs

```diff
@@ -1682,139 +1650,186 @@ public void AssemblyLoading_ResourcesInParent(AnalyzerTestKind kind)
         [CombinatorialData]
         public void AssemblyLoadingInNonDefaultContext_AnalyzerReferencesSystemCollectionsImmutable(AnalyzerTestKind kind)
         {
-            Run(kind,
-                static (AssemblyLoadContext compilerContext, AssemblyLoadTestFixture testFixture) =>
-                {
-                    // Load the compiler assembly and a modified version of S.C.I into the compiler load context. We
-                    // expect the analyzer will use the bogus S.C.I in the compiler context instead of the one 
-                    // in the host context.
-                    _ = compilerContext.LoadFromAssemblyPath(testFixture.UserSystemCollectionsImmutable);
-                    _ = compilerContext.LoadFromAssemblyPath(typeof(AnalyzerAssemblyLoader).GetTypeInfo().Assembly.Location);
-                },
-                static (AnalyzerAssemblyLoader loader, AssemblyLoadTestFixture testFixture) =>
-                {
-                    StringBuilder sb = new StringBuilder();
+            // Load the compiler assembly and a modified version of S.C.I into the compiler load context. We
+            // expect the analyzer will use the bogus S.C.I in the compiler context instead of the one 
+            // in the host context.
+            var alc = new AssemblyLoadContext(nameof(AssemblyResolver_FirstOneWins), isCollectible: true);
```

Comment:
@jjonescz this is the only test that was really using the extra `AssemblyLoadContext` and now it is just created here. Looks like I did forgot to unload it though, will fix that.